tests/integration/test_charm.py

```diff
@@ -1,32 +1,28 @@
+#!/usr/bin/env python3
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-import asyncio
 import logging
-from pathlib import Path
 
 import pytest
-import yaml
+import requests
+from conftest import deploy  # noqa: F401, pylint: disable=W0611
+from helpers import APP_NAME_AIRBYTE_SERVER, get_unit_url
 from pytest_operator.plugin import OpsTest
 
 logger = logging.getLogger(__name__)
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
-APP_NAME = METADATA["name"]
-
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
-
-    Assert on the unit status before any relations/configurations take place.
-    """
-    # Build and deploy charm from local source folder
-    charm = await ops_test.build_charm(".")
-    resources = {"httpbin-image": METADATA["resources"]["httpbin-image"]["upstream-source"]}
-
-    # Deploy the charm and wait for active/idle status
-    await asyncio.gather(
-        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME),
-        ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000),
-    )
+@pytest.mark.usefixtures("deploy")
+class TestDeployment:
+    """Integration tests for charm."""
+
+    async def test_deployment(self, ops_test: OpsTest):
+        url = await get_unit_url(ops_test, application=APP_NAME_AIRBYTE_SERVER, unit=0, port=8001)
+        logger.info("curling app address: %s", url)
+
+        response = requests.get(f"{url}/api/v1/health", timeout=300)
+        print(response.json())
+        assert response.status_code == 200
+        assert response.json().get("available")
```

Comment:
How do you tell if your worklow stuff ran successfully, I imagine you just want to check here that Airbyte is running though. Also I'm curious why Airbyte and Temporal are needed for this integration test?

Reply:
I've added another test to create and run an Airbyte connection.